tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Printers.scala

```diff
@@ -49,34 +52,54 @@ object Printers {
       case _                                         => true
     }
 
-    protected def printBlock(tree: Tree): Unit = {
-      val trees = tree match {
-        case Block(trees) => trees
-        case _            => List(tree)
+    protected def printRow(ts: List[Tree], start: String, sep: String,
+                                 end: String): Unit = {
+      print(start)
+      var rest = ts
+      while (rest.nonEmpty) {
+        print(rest.head)
+        rest = rest.tail
+        if (rest.nonEmpty)
+          print(sep)
       }
-      print("{"); indent(); println()
-      printSeq(trees) { x =>
-        printStat(x)
-      } { x =>
-        if (shouldPrintSepAfterTree(x))
-          print(";")
-        println()
+      print(end)
+    }
+
+    protected def printBlock(tree: Tree): Unit = {
+      print('{'); indent(); println()
+      tree match {
+        case tree: Block =>
+          var rest = tree.stats
+          while (rest.nonEmpty) {
+            val x = rest.head
+            rest = rest.tail
+            printStat(x)
+            if (rest.nonEmpty) {
+              if (shouldPrintSepAfterTree(x))
+                print(';')
+              println()
+            }
+          }
+
+        case _ => printStat(tree)
```

Comment:
For symmetry with the other `case`, I would put `printStat(tree)` on the next line.